src/sentry/incidents/subscription_processor.py

```diff
@@ -109,18 +109,33 @@ def __init__(self, subscription: QuerySubscription) -> None:
 
     @property
     def active_incident(self) -> Incident:
+        """
+        fetches an incident given the alert rule, project (and subscription if available)
+        """
         if not hasattr(self, "_active_incident"):
-            self._active_incident = Incident.objects.get_active_incident(
-                self.alert_rule, self.subscription.project
+            incident = Incident.objects.get_active_incident(
+                alert_rule=self.alert_rule,
+                project=self.subscription.project,
+                subscription=self.subscription,
             )
+            if not incident:
+                # TODO: make subscription required
+                incident = Incident.objects.get_active_incident(
+                    alert_rule=self.alert_rule, project=self.subscription.project
+                )
```

Comment:
No need for this conditional, we always have a subscription here since we require it to get the project

Reply:
Oh - 
we haven't made subscription required as a fk to the `Incident` model yet though

this is just covering the case where there's an incident without a subscription fk'd

Reply:
This is in the subscription processor, `self.subscription.project` is already used here

Reply:
hmmm not sure if i'm understanding you here.

In this conditional i'm making a query to `get_active_incident` first _with_ the `self.subscription` filter - and if that doesn't exist, then i make the query _without_ the subscription filter.

The check isn't for whether subscription exists in the processor (yes, this always exists)
but the check is for whether or not we've backfilled the `subscription` to the `Incident` instance yet or not

Reply:
Lol sorry, I skimmed this and didn't read the conditional properly. I thought it was `if not self.subscription`. Ignore me